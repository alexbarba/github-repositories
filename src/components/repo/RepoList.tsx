import { FC } from 'react'
import { IRepository } from 'types'
import RepoCard from './RepoCard'

type RepoCardProps = {
  repos: IRepository[]
}

const RepoList: FC<RepoCardProps> = ({ repos }) => {
  return (
    <div className="text-sm">
      <ul className="divide-y divide-accent-br">
        {repos.map((repo) => (
          <li key={repo.id}>
            <RepoCard repo={repo} />
          </li>
        ))}
      </ul>
    </div>
  )
}

export default RepoList
