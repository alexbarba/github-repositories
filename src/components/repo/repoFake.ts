export default {
  id: 377575434,
  node_id: 'MDEwOlJlcG9zaXRvcnkzNzc1NzU0MzQ=',
  name: 'aleworld-status',
  full_name: 'yknx4/aleworld-status',
  private: false,
  owner: {
    login: 'yknx4',
    id: 1848186,
    node_id: 'MDQ6VXNlcjE4NDgxODY=',
    avatar_url: 'https://avatars.githubusercontent.com/u/1848186?v=4',
    gravatar_id: '',
    url: 'https://api.github.com/users/yknx4',
    html_url: 'https://github.com/yknx4',
    followers_url: 'https://api.github.com/users/yknx4/followers',
    following_url: 'https://api.github.com/users/yknx4/following{/other_user}',
    gists_url: 'https://api.github.com/users/yknx4/gists{/gist_id}',
    starred_url: 'https://api.github.com/users/yknx4/starred{/owner}{/repo}',
    subscriptions_url: 'https://api.github.com/users/yknx4/subscriptions',
    organizations_url: 'https://api.github.com/users/yknx4/orgs',
    repos_url: 'https://api.github.com/users/yknx4/repos',
    events_url: 'https://api.github.com/users/yknx4/events{/privacy}',
    received_events_url: 'https://api.github.com/users/yknx4/received_events',
    type: 'User',
    site_admin: false
  },
  html_url: 'https://github.com/yknx4/aleworld-status',
  description:
    'ðŸ“ˆ Uptime monitor and status page for Ale Ornelas Figueroa, powered by @upptime',
  fork: false,
  url: 'https://api.github.com/repos/yknx4/aleworld-status',
  forks_url: 'https://api.github.com/repos/yknx4/aleworld-status/forks',
  keys_url: 'https://api.github.com/repos/yknx4/aleworld-status/keys{/key_id}',
  collaborators_url:
    'https://api.github.com/repos/yknx4/aleworld-status/collaborators{/collaborator}',
  teams_url: 'https://api.github.com/repos/yknx4/aleworld-status/teams',
  hooks_url: 'https://api.github.com/repos/yknx4/aleworld-status/hooks',
  issue_events_url:
    'https://api.github.com/repos/yknx4/aleworld-status/issues/events{/number}',
  events_url: 'https://api.github.com/repos/yknx4/aleworld-status/events',
  assignees_url:
    'https://api.github.com/repos/yknx4/aleworld-status/assignees{/user}',
  branches_url:
    'https://api.github.com/repos/yknx4/aleworld-status/branches{/branch}',
  tags_url: 'https://api.github.com/repos/yknx4/aleworld-status/tags',
  blobs_url:
    'https://api.github.com/repos/yknx4/aleworld-status/git/blobs{/sha}',
  git_tags_url:
    'https://api.github.com/repos/yknx4/aleworld-status/git/tags{/sha}',
  git_refs_url:
    'https://api.github.com/repos/yknx4/aleworld-status/git/refs{/sha}',
  trees_url:
    'https://api.github.com/repos/yknx4/aleworld-status/git/trees{/sha}',
  statuses_url:
    'https://api.github.com/repos/yknx4/aleworld-status/statuses/{sha}',
  languages_url: 'https://api.github.com/repos/yknx4/aleworld-status/languages',
  stargazers_url:
    'https://api.github.com/repos/yknx4/aleworld-status/stargazers',
  contributors_url:
    'https://api.github.com/repos/yknx4/aleworld-status/contributors',
  subscribers_url:
    'https://api.github.com/repos/yknx4/aleworld-status/subscribers',
  subscription_url:
    'https://api.github.com/repos/yknx4/aleworld-status/subscription',
  commits_url:
    'https://api.github.com/repos/yknx4/aleworld-status/commits{/sha}',
  git_commits_url:
    'https://api.github.com/repos/yknx4/aleworld-status/git/commits{/sha}',
  comments_url:
    'https://api.github.com/repos/yknx4/aleworld-status/comments{/number}',
  issue_comment_url:
    'https://api.github.com/repos/yknx4/aleworld-status/issues/comments{/number}',
  contents_url:
    'https://api.github.com/repos/yknx4/aleworld-status/contents/{+path}',
  compare_url:
    'https://api.github.com/repos/yknx4/aleworld-status/compare/{base}...{head}',
  merges_url: 'https://api.github.com/repos/yknx4/aleworld-status/merges',
  archive_url:
    'https://api.github.com/repos/yknx4/aleworld-status/{archive_format}{/ref}',
  downloads_url: 'https://api.github.com/repos/yknx4/aleworld-status/downloads',
  issues_url:
    'https://api.github.com/repos/yknx4/aleworld-status/issues{/number}',
  pulls_url:
    'https://api.github.com/repos/yknx4/aleworld-status/pulls{/number}',
  milestones_url:
    'https://api.github.com/repos/yknx4/aleworld-status/milestones{/number}',
  notifications_url:
    'https://api.github.com/repos/yknx4/aleworld-status/notifications{?since,all,participating}',
  labels_url:
    'https://api.github.com/repos/yknx4/aleworld-status/labels{/name}',
  releases_url:
    'https://api.github.com/repos/yknx4/aleworld-status/releases{/id}',
  deployments_url:
    'https://api.github.com/repos/yknx4/aleworld-status/deployments',
  created_at: '2021-06-16T17:24:09Z',
  updated_at: '2021-12-01T23:12:00Z',
  pushed_at: '2021-12-01T23:11:57Z',
  git_url: 'git://github.com/yknx4/aleworld-status.git',
  ssh_url: 'git@github.com:yknx4/aleworld-status.git',
  clone_url: 'https://github.com/yknx4/aleworld-status.git',
  svn_url: 'https://github.com/yknx4/aleworld-status',
  homepage: 'https://status.homie.mx',
  size: 15978,
  stargazers_count: 0,
  watchers_count: 0,
  language: null,
  has_issues: true,
  has_projects: true,
  has_downloads: true,
  has_wiki: true,
  has_pages: true,
  forks_count: 0,
  mirror_url: null,
  archived: false,
  disabled: false,
  open_issues_count: 2,
  license: {
    key: 'mit',
    name: 'MIT License',
    spdx_id: 'MIT',
    url: 'https://api.github.com/licenses/mit',
    node_id: 'MDc6TGljZW5zZTEz'
  },
  allow_forking: true,
  is_template: false,
  topics: ['status-page', 'upptime', 'uptime-monitor'],
  visibility: 'public',
  forks: 0,
  open_issues: 2,
  watchers: 0,
  default_branch: 'master'
}
