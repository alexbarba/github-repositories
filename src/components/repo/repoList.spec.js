import { render } from '@testing-library/react'
import RepoList from './RepoList'
import repoFake from './repoFake'

const dataFake = [repoFake]

describe('RepoList', () => {
  it('should be match with snapshot', () => {
    const { asFragment } = render(<RepoList repos={dataFake} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
