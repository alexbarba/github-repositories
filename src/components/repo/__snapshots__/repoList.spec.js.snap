// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList should be match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-sm"
  >
    <ul
      class="divide-y divide-accent-br"
    >
      <li>
        <div
          class="grid grid-cols-6 py-6"
        >
          <div
            class="grid col-span-5 gap-2"
          >
            <h3>
              <a
                class="mr-3 text-xl font-bold text-accent-fg hover:underline"
                href="https://github.com/yknx4/aleworld-status"
              >
                aleworld-status
              </a>
              <div
                class="border border-accent-br rounded-xl w-auto px-2 inline-block text-xs text-secondary"
              >
                Public
              </div>
            </h3>
            <p
              class="text-secondary"
            >
              ðŸ“ˆ Uptime monitor and status page for Ale Ornelas Figueroa, powered by @upptime
            </p>
            <div
              class="flex flex-wrap gap-4 my-2"
            >
              <div
                class="border border-accent-br rounded-xl w-auto px-2 inline-block text-xs bg-accent-subtle hover:bg-accent-emphasis text-accent-fg"
              >
                status-page
              </div>
              <div
                class="border border-accent-br rounded-xl w-auto px-2 inline-block text-xs bg-accent-subtle hover:bg-accent-emphasis text-accent-fg"
              >
                upptime
              </div>
              <div
                class="border border-accent-br rounded-xl w-auto px-2 inline-block text-xs bg-accent-subtle hover:bg-accent-emphasis text-accent-fg"
              >
                uptime-monitor
              </div>
            </div>
            <div
              class="inline-flex gap-4 text-xs text-secondary"
            >
              <p>
                <svg
                  aria-hidden="true"
                  class="mr-1"
                  fill="currentColor"
                  height="16"
                  role="img"
                  style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    d="M8.75.75a.75.75 0 00-1.5 0V2h-.984c-.305 0-.604.08-.869.23l-1.288.737A.25.25 0 013.984 3H1.75a.75.75 0 000 1.5h.428L.066 9.192a.75.75 0 00.154.838l.53-.53-.53.53v.001l.002.002.002.002.006.006.016.015.045.04a3.514 3.514 0 00.686.45A4.492 4.492 0 003 11c.88 0 1.556-.22 2.023-.454a3.515 3.515 0 00.686-.45l.045-.04.016-.015.006-.006.002-.002.001-.002L5.25 9.5l.53.53a.75.75 0 00.154-.838L3.822 4.5h.162c.305 0 .604-.08.869-.23l1.289-.737a.25.25 0 01.124-.033h.984V13h-2.5a.75.75 0 000 1.5h6.5a.75.75 0 000-1.5h-2.5V3.5h.984a.25.25 0 01.124.033l1.29.736c.264.152.563.231.868.231h.162l-2.112 4.692a.75.75 0 00.154.838l.53-.53-.53.53v.001l.002.002.002.002.006.006.016.015.045.04a3.517 3.517 0 00.686.45A4.492 4.492 0 0013 11c.88 0 1.556-.22 2.023-.454a3.512 3.512 0 00.686-.45l.045-.04.01-.01.006-.005.006-.006.002-.002.001-.002-.529-.531.53.53a.75.75 0 00.154-.838L13.823 4.5h.427a.75.75 0 000-1.5h-2.234a.25.25 0 01-.124-.033l-1.29-.736A1.75 1.75 0 009.735 2H8.75V.75zM1.695 9.227c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327l-1.305 2.9zm10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327l-1.305 2.9z"
                    fill-rule="evenodd"
                  />
                </svg>
                MIT License
              </p>
              <p>
                Updated 16 hours ago
              </p>
            </div>
          </div>
          <div
            class="flex flex-col justify-around col-span-1 "
          >
            <button
              class="absolute flex w-auto my-auto text-xs break-words btn"
            >
              <svg
                aria-hidden="true"
                class="w-3 h-3 mr-1 text-secondary"
                fill="currentColor"
                height="16"
                role="img"
                style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
                  fill-rule="evenodd"
                />
              </svg>
               
              <span>
                Star
              </span>
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
